gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupApi.java

```diff
@@ -20,4 +20,13 @@ public interface BackupApi {
    * @return the backupId
    */
   CompletableFuture<Long> takeBackup(long backupId);
+
+  /**
+   * Returns the status of the backup. The future fails if the request was not processed by at least
+   * one partition.
+   *
+   * @param backupId
+   * @return the status of the backup
+   */
+  CompletableFuture<BackupStatus> getStatus(long backupId);
```

Comment:
:wrench: I would recommend using `CompletionStage` wherever we want to pass an immutable future. Sorry I didn't mention it last time, but returning a `CompletableFuture` for me hints that a consumer may complete the future, which I don't think is our intended usage.

You can always switch from a CompletionStage to a CompletableFuture of course, e.g. `CompletionStage#toCompletableFuture`, but here the gain is more in semantics (strongly hinting at how the method should be used) than anything.

Reply:
:+1: Make sense. I will update both APIs to use `CompletionStage`